compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -830,6 +687,10 @@ impl<'tcx> InferCtxt<'tcx> {
         self.tcx.dcx()
     }
 
+    pub fn defining_opaque_types(&self) -> &'tcx ty::List<LocalDefId> {
```

Comment:
Since `defining_opaque_types` is a private field on `InferCtxt`, and I've chosen to expose only one trait (`SolverDelegate`) and not two (`InferCtxtLike`), we gotta make it possible to reference this in the impl in `rustc_trait_solver`.